src/darwin/Framework/CHIP/MTRDeviceController_Concrete.mm

```diff
@@ -477,6 +483,16 @@ - (void)cleanupAfterStartup
 
     [self stopBrowseForCommissionables];
 
+    // Calling stopMonitoring breaks the retain cycle of the monitor handler holding
+    // the monitor object, allowing the monitor object to dealloc and clean up.
+    {
+        std::lock_guard lock(_deviceConnectivityMonitorLock);
+        for (MTRDeviceConnectivityMonitor *deviceConnectivityMonitor in _weakSetOfDeviceConnectivityMonitors) {
+            [deviceConnectivityMonitor stopMonitoring];
```

Comment:
This is fine for now, but as a followup, could we just have the monitor pass itself to the handler?  Or would we then have issues with nothing keeping the monitor alive at all and it not working correctly?  Might be worth documenting that, if so.

Reply:
The handler is currently holding a reference to the monitor, and breaks the cycle when it's called, with `stopMonitoring`. This code here is to make sure that when a controller shuts down, we immediately clear away all the held monitors.

This is a weak set, and so in normal operation monitors clean themselves up when the handler is called.